src/kvstore/raftex/RaftPart.cpp

```diff
@@ -1472,7 +1525,7 @@ void RaftPart::processAppendLogRequest(
     }
 
     // Reset the timeout timer
-    lastMsgRecvDur_.reset();
+    helper.setLastMsgRecvDur(&lastMsgRecvDur_);
```

Comment:
I don’t think so. If the current follower processes processAppendLogRequest() for a long time, then even if the current follower becomes the leader, that may not be a good result. Because the follower that takes a long time to process processAppendLogRequest() may not be suitable for the leader.

We can analyze the reasons that cause the follower to process the processAppendLogRequest() for a long time. In my case, the disk IO is full, so if the IO is full, it will not be stable even if it becomes the leader, and the obvious suspicion in inprocessAppendLogRequest() is that Wal's use of IO may lead to longer processing and no other obvious suspicion.

Reply:
Yeah, sorry for forgetting to mention that rocksdb uses IO.
I used to see that wal and commit were called more than once in processAppendLogRequest(), and I was not sure whether each call needed to execute lastMsgRecvDur_.reset() when the call failed, because the call failure would cause processAppendLogRequest() to terminate early, so for simplicity I used RAII.

But just now I read the code again, only adding lastMsgRecvDur_.reset() at the end of processAppendLogRequest() may indeed be enough, I will test it in my environment, thanks.

Reply:
👌 , I am doing some test in 2.0 recently. You could contact me if any problem found.

Reply:
Because the SST import method does not support index updates, and there is a need for large-scale data writing in my business, it will continue to fill the disk IO for several hours (IO utils will always >=90% and often reach 100%), At this time, the leader of v1.1 will switch frequently, which seriously affects the stability of data import.

The above is my requirement, and after my current optimization, this problem almost no longer occurs. I tested that the IO is full for 20 hours (IO Utils>=90%, often 100%), and the leader change has not happened. The demand has been met now, and I hope that the stability of the leader of 2.0 will be better :P

Reply:
Yeah, I am trying to recurrent this issue. The raft part is almost exactly same with 1.1.